modules/core/src/commonMain/kotlin/fr/acinq/lightning/serialization/channel/Serialization.kt

```diff
@@ -17,9 +16,6 @@ object Serialization {
         return when {
             // v4 uses a 1-byte version discriminator
             bin[0].toInt() == 4 -> DeserializationResult.Success(fr.acinq.lightning.serialization.channel.v4.Deserialization.deserialize(bin))
-            // v2/v3 use a 4-bytes version discriminator
-            Pack.int32BE(bin) == 3 -> DeserializationResult.Success(fr.acinq.lightning.serialization.v3.Serialization.deserialize(bin))
-            Pack.int32BE(bin) == 2 -> DeserializationResult.Success(fr.acinq.lightning.serialization.v2.Serialization.deserialize(bin))
```

Comment:
This may save future headaches (otherwise it will return version 0):

```kotlin
            // v2/v3 used a 4-bytes version discriminator and are now unsupported
            Pack.int32BE(bin) == 3 -> DeserializationResult.UnknownVersion(3)
            Pack.int32BE(bin) == 2 -> DeserializationResult.UnknownVersion(2)
```